src/common

```diff
@@ -146,11 +146,11 @@ ifeq ($(origin CXX),default)
   CXX = cl
 endif
 ifeq ($(origin YACC),default)
-  YACC   = bison -y
+  YACC   = win_bison -y
```

Comment:
:pick: I am guessing this is required due to the names used by the versions of flex/bison we are using in the new job. But can you add the rationale to the commit message body so A) I don't have to guess B) Anyone who starts seeing local failures due to using different versions, knows there was a reason for doing this. Not just to ruin their day ;)

Reply:
It's worth noting here that when someone installs `winflexbison` (https://github.com/lexxmark/winflexbison) the default binaries are called `win_flex` and `win_bison`. It takes an extra manual step to rename these to `flex` or `bison`

Reply:
Is this the only place that people could possibly source flex/bison for Windows from? Putting it differently, I don't recall hearing from anyone that they were unhappy with our Makefiles for Windows, which either means that no one is using them or existing users have a working set-up. So I'm with @kroening: set `YACC='win_bison -y'` and `LEX=win_flex` on the command-line.

Reply:
There are many ways to get Flex / Bison on Windows. If a user has a build environment based on Cygwin or MSYS2 for example then the packages for those environments may still be using the original names. I'd need to check to know for sure, but I don't have a Windows machine on hand.

Reply:
@tautschnig Are we sure we want to do that? Just for clarification, there are 3 potential configurations for building on windows (that I can see in `src/common`), and they are Cygwin, MSVC and MinGW based. For them:

1. The `MinGW` configuration is inheriting `flex` and `bison` binary names from the default configuration
2. The `Cygwin` configuration also sets them explicitly to `flex` and `bison`, and
3. The `MSVC` configuration is the one I changed to address `win_flex` and `win_bison`, which are the default binaries a user should end up with if he installs `winflexbison` (which is the default way a programmer would acquire these to use them with Visual Studio today, to my understanding).

So, I guess my question is: Does this really break anything? Alternative build environments on Windows should not be affected in any capacity (if a user is using Cygwin or MinGW) and the MSVC one I can argue that was misconfigured (or at least stale by current standards).

Reply:
I have now taken another look at the rest of this file. I have now seen that there are other sets of configuration in this file for cygwin and mingw. Therefore I am less concerned about explicitly changing this for Visual Studio only. I am happy for this change to be merged as is.

Reply:
> So, I guess my question is: Does this really break anything? [...] and the MSVC one I can argue that was misconfigured [...]

I simply did not know whether `winflexbison` was the (single?) most common approach. If you're confident that's the case, then, sure, let's go with it. My concern was breaking existing user's set-ups, but if it seems unlikely that anyone was (successfully) using this configuration, then there shouldn't be a problem. Well, if there is, people will tell us and we can revert :-)